src/dune_engine/build.ml

```diff
@@ -353,79 +359,116 @@ let fold_labeled (type acc) t ~(init : acc) ~f =
     | Filter_existing_files p -> loop p acc
     | Memo m -> loop m.t acc
     | Catch (t, _) -> loop t acc
+    | Fiber _ -> acc
+    | Dyn_fiber b -> loop b acc
+    | Build b -> loop b acc
   in
   loop t init
 
 (* Execution *)
 
-module rec Execution : sig
-  val exec : 'a t -> 'a * Dep.Set.t
-end = struct
-  module Function = struct
-    type 'a input = 'a memo
-
-    type 'a output = 'a * Dep.Set.t
-
-    let name = "exec-memo"
-
-    let id m = m.id
+let do_not_use_stage_fiber f = Fiber f
```

Comment:
Do we have any expectations about what `f` can do? Can it read files, for example? I think these functions should be documented and if we place them in the `Expert` module, we should clarify what the user is expected to check before using them.

Reply:
Can you add this requirement as a comment and move all three functions into `Expert` interface?